test/benchmarks/datadog/metric_handler.py

```diff
@@ -0,0 +1,167 @@
+                else:
+                    # Log invalid tags as errors
+                    LOGGER.error(f"Tag is not a valid dataset or environment tag: tag={tag}")
+
+        return valid_tags
+
+
+class IndexingDocsPerSecond(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.indexing.docs_per_second"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingExactMatchMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.exact_match"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingF1Metric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.f1_score"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingRecallMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.recall"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingMapMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.map"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class QueryingSecondsPerQueryMetric(CustomDatadogMetric):
+    def __init__(self, value: float, tags: Optional[List[Tag]] = None) -> None:
+        name = "haystack.benchmarks.querying.seconds_per_query"
+        super().__init__(name=name, value=value, tags=tags)
+
+
+class MetricsAPI:
+    @retry(
+        retry=retry_if_exception(AssertionError),  # type: ignore
+        wait=wait_fixed(5),
+        stop=stop_after_attempt(3),
+        reraise=True,
+    )
+    def send_custom_dd_metric(self, metric: CustomDatadogMetric) -> dict:
+        options = {
+            "api_key": os.getenv("DD_API_KEY"),
+            "app_key": os.getenv("DD_APP_KEY"),
+            "api_host": os.getenv("DD_HOST"),
+        }
+
+        datadog.initialize(**options)
+        tags: List[str] = list(map(lambda t: str(t.value), metric.tags))
+
+        post_metric_response: Dict = datadog.api.Metric.send(
+            metric=metric.name, points=[metric.timestamp, metric.value], tags=tags
+        )
+
+        if post_metric_response.get("status") != "ok":
+            LOGGER.error(
+                f"Could not send custom metric. metric_name={metric.name}, metric_value={metric.value}, status={post_metric_response.get('status')}, error={post_metric_response.get('errors')}, {post_metric_response}"
+            )
+        else:
+            LOGGER.info(
+                f"Sent custom metric. metric_name={metric.name}, metric_value={metric.value}, status={post_metric_response.get('status')}"
+            )
+
+        assert (
```

Comment:
Library code should never assert (in general, should never exit the process). You should return a meaningful error 
/ raise an exception that the caller code can handle (and eventually exit the process)